common/djangoapps/student/views/management.py

```diff
@@ -1185,8 +1186,11 @@ def password_reset_confirm_wrapper(request, uidb36=None, token=None):
         password = request.POST['new_password1']
 
         try:
+            if not isinstance(password, text_type):
+                # some checks rely on unicode semantics (e.g. length)
+                password = text_type(password, encoding='utf8')
```

Comment:
You do this same code in three places, right before calling validate_password. Maybe move that combo to a utility function in this file?